File: practica2_820574_839304/java/Push.java
/*
* Fichero: Push.java
* Autores: Alicia Lázaro Huerta y Manel Jordá Puig Rubio
* Fecha: 27/02/2023
*/
import java.util.Stack;

public class Push extends Instruccion{
    private int valor;
    Push(int num){
        nombre = "push";
        valor = num;
    }
    public void ejecutar(Stack<Integer> pila, int[] pc){
        pila.push(valor);
        pc[0]++;
    }
}

File: practica2_820574_839304/c++/push.cc
/*
* Fichero: push.cc
* Autores: Alicia Lazaro Huerta y Manel Jorda Puig Rubio
* Fecha: 27/02/2023
*/

#include "push.h"

Push::Push(int _num) {
    nombre = "push";
    num = _num;
}

void Push::ejecutar(stack<int> &pila, int &pc) {
    pila.push(num);
    pc++;
}

string Push::nombrar() {
    return nombre;
}